src/lib.rs

```diff
@@ -11,13 +11,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! A failpoints implementation for rust.
+//! A fail points implementation for rust.
+//!
+//! Fail points are code points that used to inject error by user at runtime.
+//! This crate is inspired by FreeBSD's [failpoints](https://freebsd.org/cgi/man.cgi?query=fail).
+//! Unlike FreeBSD's implementation, this crate only support configuration from
+//! environment variables and API.
+//!
+//! ## Installation
+//!
+//! Add this to your `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! fail = "0.1"
+//! ```
+//!
+//! ## Example
+//!
+//! ```rust
+//! #[macro_use]
+//! extern crate fail;
+//!
+//! fn f1() {
+//!     fail_point!("example");
+//!     panic!();
+//! }
+//!
+//! fn main() {
+//!    fail::setup();
+//!    fail::cfg("rust_out::example", "return").unwrap();
+//!    f1();
+//!    fail::teardown();
+//! }
+//! ```
+//!
+//! The above example defines a fail point named "example". And then configure it as `return`.
+//! So function `f1` will return early and never panic. You can also configure it via environment
+//! variable `FAILPOINTS=rust_out::example=return`. For more supported configuration, please see
```

Comment:
You can also configure it via the `FAILPOINTS=rust_out::example=return` environment variable.

Reply:
-> For more supported configuration, see